kadasrouting/gui/navigationpanel.py

```diff
@@ -160,6 +163,50 @@ def getInstructionsToWaypoint(waypoint, gpsinfo):
             "eta": eta_string}
 
 
+class NavigationFromWaypointsLayer():
+    def __init__(self):
+        self.crs = QgsCoordinateReferenceSystem(4326)
+        itemRegex = {
+            'KadasGpxRouteItemRegex': r'^<MapItem(.*)name="KadasGpxRouteItem"(.*)CDATA(.*)]><\/MapItem>',
+            'KadasGpxWaypointItemRegex': r'^<MapItem(.*)name="KadasGpxWaypointItem"(.*)CDATA(.*)]><\/MapItem>'
+        }
+        # parse project
+        with ZipFile(QgsProject.instance().fileName()) as qgzPrj:
```

Comment:
This considers the case of the project being saved as a qgz file (the default one), but not as a qgs file, which is still possible from the UI

Reply:
good catch, I will check for the extension.

Reply:
As I implemented the fact that we save a tmp file, I have control on this. So I use explicitly a qgz file